style.md

```diff
@@ -1923,6 +1924,90 @@ sptr := &T{Name: "bar"}
 </td></tr>
 </tbody></table>
 
+### Initializing Maps
+
+Prefer `make(..)` for empty maps, and maps populated
+programmatically. This makes map initialization visually
+distinct from declaration, and it makes it easy to add size
```

Comment:
I went down a similar path before pushing the initial PR and decided not to - the tl;dr being that we should probably have some baseline expectation of the audience's familiarity with Go, particularly the behavior of builtin types.

Totally up to revisit that assumption, or for us to officially set a baseline of expected knowledge of the language - e.g. "for Go beginners", "for intermediate Go users", "for advanced Go users", "for anyone with some experience writing software", etc. What do you think?

Reply:
Good question, I haven't really thought about a specific audience other than Go engineers at Uber. I'd say mostly beginner/intermediate?

In this case, it might be easy to miss that declaration = `nil` map, but since we call out the `m2 will panic on writes` below, it might be less important to call this out here.

Reply:
Yeah, it'd definitely be easier to expand on this as we find that folks are confused by it.